apps/api/src/routes/gas-sponsorship.ts

```diff
@@ -0,0 +1,74 @@
+import type { FastifyPluginAsync } from "fastify";
+
+import type { ErrorReply } from "../schema";
+import {
+  type ValidateBody,
+  type ValidateParams,
+  type ValidateReply,
+  validateBodySchema,
+  validateReplySchema,
+} from "../schema/gas-sponsorship";
+import type { TdkApiContext } from "../types";
+
+// TODO: Replace with actual sponsored partner IDs or logic to fetch them from another service like TMC
+const fullySponsoredPatnerIds = new Set(["zeeverse", "smols"]);
+
+export const gasSponsorshipRoutes =
+  ({ db }: TdkApiContext): FastifyPluginAsync =>
+  async (app) => {
+    app.post<{
+      Params: ValidateParams;
+      Body: ValidateBody;
+      Reply: ValidateReply | ErrorReply;
+    }>(
+      "/gas-sponsorship/:partnerId/validate",
```

Comment:
Please follow a [RESTful or resource-based endpoint design pattern](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design) as closely as possible in the TDK API. This endpoint would instead be something like `POST /partners/:partnerId/gas-sponsorship`